mempool/clist_mempool.go

```diff
@@ -47,11 +48,21 @@ type CListMempool struct {
 	// Keeps track of the rechecking process.
 	recheck *recheck
 
-	// Data in `txs` and `txsMap` must to be kept in sync and updated atomically.
+	// Data in the following variables must to be kept in sync and updated atomically.
 	txsMtx   cmtsync.RWMutex
-	txs      *clist.CList                    // concurrent linked-list of valid txs
-	txsMap   map[types.TxKey]*clist.CElement // for quick access to txs
+	lanes    map[types.Lane]*clist.CList     // each lane is a linked-list of (valid) txs
+	txsMap   map[types.TxKey]*clist.CElement // for quick access to the mempool entry of a given tx
 	txsBytes int64                           // total size of mempool, in bytes
+	numTxs   int64                           // total number of txs in the mempool
+
+	addTxChMtx    cmtsync.RWMutex // Protects the fields below
```

Comment:
Do we really need two mutexes here? (this one and the one in line 52). It the fact of having two mutexes for performance reasons? If yes, are we sure there's no risk of deadlock when using them?

Reply:
@hvanz  has a draft PR that should remove them. However, we discussed to try and merge this first and then have a new PR to address this potential issue. Especially since that code is based of a slightly older version of this branch so it takes some time to redesign it atm.